File: src/main/java/cat/xlagunas/andrtc/user/UserService.java
package cat.xlagunas.andrtc.user;

import org.springframework.security.core.userdetails.UserDetailsService;

import java.util.List;

public interface UserService extends UserDetailsService {

    UserDto findUser(long idUser) throws UserNotFoundException;

    UserDto findUser(String username) throws UserNotFoundException;

    List<UserDto> searchByUsername(String username);

    void createUser(UserDto userDto) throws ExistingUserException;

    boolean updatePassword(long userId, String password);

    boolean updateProfilePicture(long userId, String profileUrl);

}

File: src/main/java/cat/xlagunas/andrtc/user/UserService.kt
package cat.xlagunas.andrtc.user

import org.springframework.security.core.userdetails.UserDetailsService

interface UserService : UserDetailsService {

    @Throws(UserNotFoundException::class)
    fun findUser(idUser: Long): UserDto

    @Throws(UserNotFoundException::class)
    fun findUser(username: String): UserDto

    fun searchByUsername(username: String): List<UserDto>

    @Throws(ExistingUserException::class)
    fun createUser(userDto: UserDto)

    fun updatePassword(userId: Long, password: String): Boolean

    fun updateProfilePicture(userId: Long, profileUrl: String): Boolean

}
